public/components/visualization/text2viz.tsx

```diff
@@ -111,14 +127,23 @@ export const Text2Viz = () => {
           });
         } else {
           setEditorInput(JSON.stringify(result, undefined, 4));
+
+          // Report metric when visualization generated successfully
+          if (usageCollection) {
```

Comment:
Could use `METRIC_TYPE.COUNT` to either record a 1 for success or 0 for failure. We could calculate average to get success rate.

Reply:
Currently, it doesn't record the failures. If we do need to record the failures, I'd rather use another event type, like:
```typescript
  usageCollection.reportUiStats(
    VIS_NLQ_APP_ID,
    usageCollection.METRIC_TYPE.LOADED,
    `generate_failed-${uuidv4()}`
  );
```
What do you think?

Reply:
I mean we could use the same metric, and emit 0 when fails, 1 when succeeds. Like
```
try {
 const response = await fetch();
emitMetric("fetch_success", COUNT, 1);
} catch( err: any) {
emitMetric("fetch_success", COUNT, 0);
}
```
We always ensure that we emit one time for a fetch, it's either 1 or 0.